extensions/amp-a4a/0.1/amp-ad-network-base.js

```diff
@@ -68,7 +68,9 @@ export class AmpAdNetworkBase extends AMP.BaseElement {
 
   /** @override */
   onLayoutMeasure() {
-    this.sendRequest_();
+    if (!this.adResponsePromise_) {
```

Comment:
```  
sendRequest_() {
    Services.viewerForDoc(this.getAmpDoc())
      .whenFirstVisible()
      .then(() => {
        const url = this.getRequestUrl();
        this.adResponsePromise_ = sendXhrRequest(this.win, url);
      });
  }
```

This above code can still have race condition. Multiple requests can be sent while waiting for the `whenFirstVisible`.

I'd suggest do the following:

```js
sendRequest_() {
  return ...
}

onLayoutMeasure() {
  if (!adResponsePromise_) {
    adResponsePromise = this.sendRequest_();
  }
}
```

note you will need to change the handleFailure code as well

Reply:
Done.